sql/activesets/activesets.go

```diff
@@ -50,25 +54,28 @@ func Get(db sql.Executor, id types.Hash32) (*types.EpochActiveSet, error) {
 }
 
 func GetBlob(db sql.Executor, id []byte) ([]byte, error) {
-	var rst []byte
-	rows, err := db.Exec("select active_set from activesets where id = ?1;",
-		func(stmt *sql.Statement) {
-			stmt.BindBytes(1, id)
-		},
-		func(stmt *sql.Statement) bool {
-			rd := stmt.ColumnReader(0)
-			rst = make([]byte, rd.Len())
-			rd.Read(rst)
-			return true
-		},
-	)
-	if rows == 0 {
-		return nil, fmt.Errorf("active set %x: %w", id, sql.ErrNotFound)
-	}
-	if err != nil {
-		return nil, fmt.Errorf("get active set blob %x: %w", id, err)
-	}
-	return rst, nil
+	cacheKey := sql.QueryCacheKey(CacheKindActiveSetBlob, string(id))
+	return sql.WithCachedValue(db, cacheKey, func() ([]byte, error) {
+		var rst []byte
+		rows, err := db.Exec("select active_set from activesets where id = ?1;",
+			func(stmt *sql.Statement) {
+				stmt.BindBytes(1, id)
+			},
+			func(stmt *sql.Statement) bool {
+				rd := stmt.ColumnReader(0)
+				rst = make([]byte, rd.Len())
+				rd.Read(rst)
+				return true
+			},
+		)
+		if rows == 0 {
+			return nil, fmt.Errorf("active set %x: %w", id, sql.ErrNotFound)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("get active set blob %x: %w", id, err)
```

Comment:
same

Reply:
same as above, could probably make a `types.Hash32` out of it, but not sure it's worth it